acceptance/acceptance_test.go

```diff
@@ -229,1905 +99,1417 @@ func testWithoutSpecificBuilderRequirement(
-			h.AssertMatch(t, output, `Paketo Buildpacks:\s+'gcr.io/paketo-buildpacks/builder:full-cf'`)
-			h.AssertMatch(t, output, `Paketo Buildpacks:\s+'gcr.io/paketo-buildpacks/builder:tiny'`)
+			output := pack.SuccessfulRunWithOutput("suggest-builders")
+
+			assertOutput := assert.NewOutputAssertionManager(output)
+			assertOutput.IncludesSuggestedBuildersHeading()
+			assertOutput.IncludesGoogleBuilder()
+			assertOutput.IncludesHerokuBuilder()
+			assertOutput.IncludesPaketoBuilders()
 		})
 	})
 
 	when("suggest-stacks", func() {
 		it("displays suggested stacks", func() {
-			cmd := subjectPack("suggest-stacks")
-			output, err := cmd.CombinedOutput()
-			if err != nil {
-				t.Fatalf("suggest-stacks command failed: %s: %s", output, err)
-			}
-			h.AssertContains(t, string(output), "Stacks maintained by the community:")
+			output := pack.SuccessfulRunWithOutput("suggest-stacks")
+
+			assert.NewOutputAssertionManager(output).IncludesSuggestedStacksHeading()
 		})
 	})
 
 	when("set-default-builder", func() {
 		it("sets the default-stack-id in ~/.pack/config.toml", func() {
-			output := h.Run(t, subjectPack("set-default-builder", "gcr.io/paketo-buildpacks/builder:base"))
-			h.AssertContains(t, output, "Builder 'gcr.io/paketo-buildpacks/builder:base' is now the default builder")
+			builderName := "gcr.io/paketo-buildpacks/builder:base"
+
+			output := pack.SuccessfulRunWithOutput("set-default-builder", builderName)
+
+			assert.NewOutputAssertionManager(output).ReportsSettingDefaultBuilder(builderName)
 		})
 	})
 
 	when("trust-builder", func() {
 		it("sets the builder as trusted in ~/.pack/config.toml", func() {
-			h.SkipIf(t, !packSupports(packPath, "trust-builder"), "pack does not support 'trust-builder'")
+			h.SkipIf(t, !pack.Supports("trust-builder"), "pack does not support 'trust-builder'")
 			builderName := "some-builder" + h.RandString(10)
 
-			h.Run(t, subjectPack("trust-builder", builderName))
+			pack.SuccessfulRun("trust-builder", builderName)
 
-			packConfigFileContents, err := ioutil.ReadFile(filepath.Join(packHome, "config.toml"))
-			h.AssertNil(t, err)
-			h.AssertContains(t, string(packConfigFileContents), builderName)
+			packConfigFileContents := pack.FileContentsInHome("config.toml")
+			assert.Contains(packConfigFileContents, builderName)
 		})
 	})
 
 	when("package-buildpack", func() {
-		var tmpDir string
+		var (
+			tmpDir                  string
+			simplePackageConfigPath string
+			containerManager        managers.TestContainerManager
+		)
 
 		it.Before(func() {
 			h.SkipIf(t,
-				!packSupports(packPath, "package-buildpack"),
+				!pack.Supports("package-buildpack"),
 				"pack does not support 'package-buildpack'",
 			)
 
-			h.SkipIf(t, dockerHostOS() == "windows", "These tests are not yet compatible with Windows-based containers")
+			containerManager = managers.NewTestContainerManager(t, assert, dockerCli)
```

Comment:
Curious why we're instantiating `containerManager` here instead of passing it in at the suite level?

Reply:
We don't need it for every test, so when focusing on tests that don't use containers we shouldn't instantiate this.

The test object we have in our test examples is different to the one we have before `spec` invokes our test commands, so if you instantiate anything with `t` in `TestAcceptance` and then pass it in to `testAcceptance` via `suite` it won't include the same `t` that is also passed into `testAcceptance`. This leads to inconsistent output and won't include the tracing of which test case is being executed.